frontend/src/metabase-lib/v1/expressions/clause.ts

```diff
@@ -49,3 +58,24 @@ export function getMBQLName(
   // case-insensitive
   return EXPRESSION_TO_MBQL_NAME.get(expressionName.trim().toLowerCase());
 }
+
+export const clausesForMode = memoize(
```

Comment:
This is just refactoring. This helper was implemented in some form in a few different places.